src/rgen/Microsoft.Macios.Generator/Emitters/BindingSyntaxFactory.ObjCRuntime.cs

```diff
@@ -362,6 +362,62 @@ static partial class BindingSyntaxFactory {
 		return LocalDeclarationStatement (declaration);
 	}
 
+	internal static LocalDeclarationStatementSyntax? GetNSValueAuxVariable (in Parameter parameter)
+	{
+
+		// the BindFrom attribute with a nsvalue supports the following types:
+		// - CGAffineTransform
+		// - NSRange
+		// - CGVector
+		// - SCNMatrix4
+		// - CLLocationCoordinate2D
+		// - SCNVector3
+		// - SCNVector4
+		// - CGPoint / PointF
+		// - CGRect / RectangleF
+		// - CGSize / SizeF
+		// - UIEdgeInsets
+		// - UIOffset
+		// - MKCoordinateSpan
+		// - CMTimeRange
+		// - CMTime
+		// - CMTimeMapping
+		// - CATransform3D
+		var t = parameter.Type.Name;
+
+		// get the factory method based on the parameter type, if it is not found, return null
+		var factoryMethod = parameter.Type switch { { FullyQualifiedName: "CoreGraphics.CGAffineTransform" } => "FromCGAffineTransform", { FullyQualifiedName: "Foundation.NSRange" } => "FromRange", { FullyQualifiedName: "CoreGraphics.CGVector" } => "FromCGVector", { FullyQualifiedName: "SceneKit.SCNMatrix4" } => "FromSCNMatrix4", { FullyQualifiedName: "CoreLocation.CLLocationCoordinate2D" } => "FromMKCoordinate", { FullyQualifiedName: "SceneKit.SCNVector3" } => "FromVector", { FullyQualifiedName: "SceneKit.SCNVector4" } => "FromVector", { FullyQualifiedName: "CoreGraphics.CGPoint" } => "FromCGPoint", { FullyQualifiedName: "CoreGraphics.CGRect" } => "FromCGRect", { FullyQualifiedName: "CoreGraphics.CGSize" } => "FromCGSize", { FullyQualifiedName: "UIKit.UIEdgeInsets" } => "FromUIEdgeInsets", { FullyQualifiedName: "UIKit.UIOffset" } => "FromUIOffset", { FullyQualifiedName: "MapKit.MKCoordinateSpan" } => "FromMKCoordinateSpan", { FullyQualifiedName: "CoreMedia.CMTimeRange" } => "FromCMTimeRange", { FullyQualifiedName: "CoreMedia.CMTime" } => "FromCMTime", { FullyQualifiedName: "CoreMedia.CMTimeMapping" } => "FromCMTimeMapping", { FullyQualifiedName: "CoreAnimation.CATransform3D" } => "FromCATransform3D",
```

Comment:
split in multi lines ?

Reply:
Ugh, that was the formatter we use :/ It was in multiple lines. Thanks for catching that.